### DISCUSSION TITLE
I request support for Winbond W25R256JWEQ / 1,7 V - 1,95 V

### DISCUSSION THREAD

**Discussion by KevinMosquera94** [nofeletru/UsbAsp-flash#107] [Q&A]
Hello,

AsProgrammer 2.0.4 does not recognize the W25R256JWEQ 1.8v connected via CH341A at 1.8v and cannot be written to.
@nofeletru Do you think support could be added please? If not, what do you use to flash the W25R256JW?

I would appreciate your help since it is to write a BIOS for an ASUS ROG ALLY

![(1)](https://github.com/nofeletru/UsbAsp-flash/assets/157324627/4fc2f013-d4f5-41fa-b423-62a0c737d240)
![PM](https://github.com/nofeletru/UsbAsp-flash/assets/157324627/85997cf2-fb1a-4466-a889-1acedf4e27ba)

**Comment by mfoster978** [comment]
Did you ever find an answer? im stuck in the same boat with that exact bios chip but on a GPD win 4 2023. Ive been looking everywhere and im trying to avoid sending it back to the manufacturer. Hopefully you get this message soon.

**Reply by astearon** [reply]
Same here, i have the same issue with the same GPD Win4 2023

**Reply by Torrengen** [reply]
Just fixed mine, i used NeoProgrammer instead :D did a whole writeup on the GPD Discord (in win_4_general chat)
Make sure u got a 1.8V adapter though, so u dont fry it :)

**Reply by NGOTRUNG6191** [reply]
@Torrengen Can you share your NeoProgrammer download link? I can find it on the internet but never have the same version that supports Winbond W25R256JWEQ / 1,7 V - 1,95 V chip :( Please help :((

**Reply by Torrengen** [reply]
@NGOTRUNG6191 here u go :) 
https://khandishnetwork.com/dl/neoprogrammer-new-update-v2-2-0-10-15-10-2021/

**Reply by NGOTRUNG6191** [reply]
> @NGOTRUNG6191 here u go :) https://khandishnetwork.com/dl/neoprogrammer-new-update-v2-2-0-10-15-10-2021/

OMG, you are a life saver, thank you very much. 
My CH341 going to arrived soon, I gonna try and flash it, thank you

**Comment by Joly0** [comment]
Is there an answer? I am also trying to find a way to get reflash the bios of my amd framework mainboard

**Comment by mfoster978** [comment]
Unfortunately I have not found any answers. I ended up shipping my gpd back
to Hong Kong it just delivered today. They said they would cover it under
warranty but we will see how long it takes for them to repair it. Huge
disappointment. This chip is too new I couldn't find any programs that
would let me write to it.

**Comment by mfoster978** [comment]
Yup, I'm still waiting to get mine back right now. They have had it for
about a month but said they are going to move it up the repair list to
prioritize it. Man I miss mine. It's been too long without it.

**Reply by astearon** [reply]
I found a programmer that supports the BIOS chip they use, it is called REVELPROG-IS and on the producer site it mentions support for W25R256JW. I will try that one and return with my findings.

The other solution i found is ask GPD for an already programmed chip and ship it my way. That will be my backup plan.

**Reply by Torrengen** [reply]
Seems u need their programmer to use their software though, i made it work with NeoProgrammer instead :)

**Reply by astearon** [reply]
Their software is free to download, i didn't mind, i got the spare chips and successfully flashed one and verified it matches the rom file. However after replacing the chip i still get no boot and no power led. I might end up sending it to GPD in the end.

GPD did reply and they are sending a flashed chip my way but i got little hopes that one will work.

**Reply by Torrengen** [reply]
Ah yeh it shouldn't matter as long as they make contact with the chip and manage to read/write and verify :) Did u make sure the BIOS u flashed was the correct one? that's what made me brick my GPD WIN 4 in the first place, turns out they got a different one for my 6800u version xD 

No Power LED sounds like a power issue though, i still got the LED even on a bricked device, so u might have to let it sit in the charger for a while first... either way, good luck with the flashed chip!

**Reply by astearon** [reply]
my power issue happened right after the bios finished flashing and tried to restart.
I have the 2023 version so 7840U one might behave differently

**Comment by mfoster978** [comment]
Yeah probably better than sending your entire GPD to them. It's been a long
time without mine.

Appreciate the info though sure it will help someone else here.

**Comment by Joly0** [comment]
Btw guys, i have resd somewhere, that you can just edit a config file or something, and it should work with the controller

**Reply by astearon** [reply]
If you can find the exact tutorial then please share, i have not been able to write to it so far

**Reply by Joly0** [reply]
Yes, i have been searching. I found it once, wanted to try it, but couldnt at that time and now i cant find that anymore. 
I know it was in a forum somewhere. The op asked about a winbond chip (not sure if it was exactly this one here), but someone answered, that you can simply search for an equivalent chip thats supported, add a line about the new chip in a config file, with the info of the equivalent chip, and that should work. In case of the forum op, it worked

**Reply by Joly0** [reply]
Btw, i have found the forum, its here https://www.mikrocontroller.net/topic/558910 but its german

**Comment by mfoster978** [comment]
I just read this and got excited.

Damn.

Well, I can tell you so far it's been maybe a month and a half since I sent
mine to gpd. 3 days ago they sent me a picture of the Win 4 turned on and
at the windows login and asked for my address to send it back.

Just thought I'd let you know, that's where I'm at so far.

**Reply by Torrengen** [reply]
Nice! as long as it gets fixed, i guess it doesnt matter how :D 
It was personally way cheaper and quicker for me to buy the components and fix it myself though, than sending it across the world and back for a repair ^^"

**Comment by mfoster978** [comment]
That makes sense. I have the 2023 version and flashed the 2022 version
bios. My power light just shut off completely. Tried leaving it plugged in
with no luck. Sent it in to GPD and waiting for a response but I can ask
them for sure if there was a power issue or if it was the bios chip. My
guess is the bios was the issue but it's possible I'm wrong. I'll try to
get some more info on that.

**Comment by mfoster978** [comment]
Yeah I don't blame you. I would have preferred to fix it myself like you
but couldn't get the things to flash with the bios chip being new it's
difficult finding any info on it. I'm interested to know how you got it
flashed and when the power light still isn't on. Does make me wonder if the
power really is an issue here. I do remember after the flash the light was
on and the fan was at full blast I waited a good 30 minutes thinking it was
still doing something. Eventually shut it off and that's when the power
light was off and I couldn't get it back on again. So the flashing itself
didn't fry out anything as if the bios wasn't fully wiped until I turned it
off.

**Reply by Torrengen** [reply]
indeed, that's basically how mine behaved, though i could get it to spin back up if i held the reset button and power button in for a few secs, it always just shut itself off again before anything showed on the screen etc. At least the charging & power circuitry seems to be unconnected to the BIOS so it wasnt much risk of an explosive failure, but it still sucked to have a 1200€ paperweight for months...

The full write-up is on their Discord, but the TL:DR; version is i bought the CH341a programmer, a 1.8V adapter and a WSON 8x6mm pogo pin prober, and eventually managed to read the chip and re-program the Flash with the 32.786 kb file found in their BIOS driver :) The main issue was finding a software that recognized it since it seems to be a newer chip with less support so far, but after i found one that connected, it was surprisingly simple (though i had to hold the prober very still for 10 mins straight while it flashed it xD)

**Reply by trillox9** [reply]
Hi, you're talking about the 25R256JWEQ, right? What software did you find to be able to flash it? I can't find anything, could you help me? Thank you @Torrengen

**Reply by Torrengen** [reply]
yes, as i mentioned earlier in this post, i used NeoProgrammer, here's the link :)
> @NGOTRUNG6191 here u go :) https://khandishnetwork.com/dl/neoprogrammer-new-update-v2-2-0-10-15-10-2021/

**Reply by trillox9** [reply]
> yes, as i mentioned earlier in this post, i used NeoProgrammer, here's the link :)
> 
> > @NGOTRUNG6191 here u go :) https://khandishnetwork.com/dl/neoprogrammer-new-update-v2-2-0-10-15-10-2021/

On Neoprogrammer, there is the BIOS 25Q256JW, but not the 25R256JWEQ. Are they the same thing? Also, did you need the Pogo WSON8 8x6 to read this BIOS?

**Comment by mfoster978** [comment]
Hmm never tried the power button and reset at the same time so I can't
confirm on my end.

The whole situation just sucks.

Question, can you post that discord link here so others can see it?

Maybe someone else can get a little further possibly would be nice to have
a solution other than sending it to GPD to get fixed.

A $1200 paperwork is not what anyone wants lol at this point I just can't
Wait to get mine back at this point. Using my gaming laptop in the meantime
for some reason isn't better even with much higher specs hahahaha

**Reply by Torrengen** [reply]
it's the first one that pops up on Google, dunno how permanent they are but this one works at least:
https://discord.com/invite/gpd-devices-243411108940087297

No kidding about finding info on this tho, even went as far as google translating random russian and german websites xD but trial n error prevailed, once i had the right equipment ^^

**Comment by mfoster978** [comment]
LOL I did the same thing too 🤣




For anyone that wants the directions from Discord I posted it below:

BTW figured id give some more info about un-bricking the bios flash:
First, i had to wait ages for 3 things to arrive in the mail:
- 1 flash programmer, namely CH341a (i got the B variant but i think its
just a bugfixed version, it uses the same drivers etc without issue)
- a 1.8V converter, seeing as the BIOS chip used was a Winbond 25R256JWEQ
- 8x6mm WSON-8 pogo pin prober, buy 2 just in case one breaks/is DOA,
they're delicate.

Get the CH341 drivers from their own website here:
https://www.wch-ic.com/downloads/CH341SER_ZIP.html

There's aparently a bunch of Flash programmer softwares out there, one guy
on Reddit (
https://www.reddit.com/r/gpdwin/comments/v80t43/success_unbricking_gpd_win_max_2021/?rdt=42318
) recommended ASProgrammer, but their newest version didn't support my
chip, so i used NeoProgrammer v2.2.0.10 (24.05.2024) myself.

Hook it all up, make sure the fuse on the CH341a bridges the 1 & 2 (this
selects progamming from a PC, apparently), and the 1.8V converter is locked
into the 25xx slot (closest to the USB port). Make sure to note the
orientation by the pin and notch.

Find the BIOS chip, mine was luckily next to the CPU and RAM here ->
Double-check that all the cables are unhooked, especially the battery (i
put a piece of tape between it to make sure)

Before you start probing, verify that the programmer's red LED is on and
bright, and you have the new BIOS handy. I got mine from downloading
https://gpd.hk/gpdwin4firmwaredriver -> Drivers & BIOS -> (New) WIN 4 BIOS
V3.06, and selecting WIN4_19_0606A/G1618-04_V3.06.rom (32 786kb).

Make sure your hand is resting on something, then start prodding the chip,
making sure all the pins line up and make contact, i struggled with this
for a good while before making it work.

Once you do, keep clicking the "Detect IC" button while jiggling the probes
a bit between each run untill it shows up.

**Reply by Torrengen** [reply]
u did forget the 2nd part tho xD

> When it shows up, the real fun starts, as now you're gonna hold that position perfectly for like 10 minutes straight.
> 
> First Read the IC, to make sure you have good contact. This can take about 5 mins. When it finishes, click Verify, which shouldnt take nearly as long.
> 
> If it's all successful, open the new BIOS file, and make sure all these options are selected: 
> Off-protect, Erase, Write, Verify. (don't need a Blank Check)
> 
> 
> Now, write to the IC, and be patient. After 5-10 minutes it should all be done with a green Success from the verification.
> 
> Then it's time to hook everything up, make sure the cooling system is screwed in (i personally just held it on with my fingers for a test boot), and pray to whatever diety you believe in 😄When it shows up, the real fun starts, as now you're gonna hold that position perfectly for like 10 minutes straight.

![image](https://github.com/nofeletru/UsbAsp-flash/assets/84272559/bd3f8481-5c75-4982-b5d1-9ddba88a2c14)

**Reply by NGOTRUNG6191** [reply]
@mfoster978 Can you share your NeoProgrammer download link? I can find it on the internet but never have the same version that supports Winbond W25R256JWEQ / 1,7 V - 1,95 V chip :( Please help :((

**Reply by trillox9** [reply]
True, I also tried several versions of NeoProgrammer, but none support the 25R256JWEQ. @mfoster978 @Torrengen  Could you please help?

**Comment by mfoster978** [comment]
Oops lol thank you. Hopefully others here might have a working solution. I
mean mines fixed but if for some reason it happens again if like to be able
to fix it myself you know.

**Comment by mfoster978** [comment]
I have the 7840u version. Figured that was late 2023 version.

**Reply by astearon** [reply]
i corrected my post , i have the same one ... 64bit 4tb, i might try flashing another chip, test read it a few times and then put it on the board.
i don't have a power draw indicator but i have a cable that lights up when connected to a usb device that takes a charge.
I saw the cable does light up and after a few seconds it turns of and on again, only once.
So there are some signs of life but not much.... tried re-seating the chip but to no avail.
I have 4 more spares so i still have a few shots left.
I will also try and read the old chip to see if it burnt itself out after the botched flash.
If that one is dead then there are chances other resistors/caps or other components near the chip also burned out.... and in that case i have no choice but to have it fixed by GPD

**Comment by mfoster978** [comment]
Either way please keep us posted I'm curious. If I could do some testing
myself I would but for obvious reasons I can't. There has to be a way to
get it working.

I'm curious if the original bios chip burned out myself.

Still waiting on a response from GPD on what happened with mine but I won't
hear back from them until probably Monday or Tuesday.

**Reply by astearon** [reply]
tried the old chip, it was not burned out.
uploaded the firmware and triple checked for read errors, all is ok but still not working when placed back on the board.
perhaps the older bios messed some other chip/chips as well?

**Reply by astearon** [reply]
The weird thing is ... now that i think about it... is that the firmware that was already on the chip looked to be the same one i was uploading.
So my initial trial of writing the chip directly on the board using CH341A was successful (meaning it uploaded in about 10 minutes and also checked if it matches the original file)
This confuses me further. Either the old firmware that I flashed went on a complete other chip on the board or something is more screwy than I thought.

**Reply by Torrengen** [reply]
did you do a straight write, or did you select Unlock->Erase->Write->Verify? as for which chip, it should be the correct one if it's a Winbond 25xNNN smth, where the NNN should match the size of your BIOS file :) 
For reference, here's mine on the 6800u (though the newer WIN 4 has it on the back by the left trigger).
![20231217_195311](https://github.com/nofeletru/UsbAsp-flash/assets/84272559/31a9e646-d109-4e1e-98ca-aea4ea5aa5e4)

**Reply by astearon** [reply]
![photo_2024-05-07_18-04-38](https://github.com/nofeletru/UsbAsp-flash/assets/7342330/24ae555b-adab-480e-a4dc-bcb543a40a62)
Here is the chip i have, it might be the same as yours.
I've used this to program, read, verify and compare:
[RevelProgrammer](https://www.reveltronics.com/en/products/revelprog-is-serial-memory-programmer-usb)
It comes with it's own software named: REVELPROG-IS
I used this one because it can erase, write and verify in about 2 minutes.
I even erased and verified manually after it was done with its own auto process.

**Reply by astearon** [reply]
Tried to upload the older firmware with the same result, chip reads fine...erased...written and verified twice

**Comment by mfoster978** [comment]
Okay so I just got word the GPD was delivered back to my house but won't be
home for a couple days so my neighbor grabbed it for me.

Anyways I sent an email to Sophie asking exactly what went bad. I know some
of us are thinking it's more than just the chip for those that had similar
issues as me so I figured it would be helpful if she talked to the repair
guy and got back with me.

Ill keep you posted.

**Reply by astearon** [reply]
If you do have a programmer please try and get the bios firmware they uploaded...having a known good is always nice to have.
Also, can you confirm what model you have?

**Comment by mfoster978** [comment]
The programmer I have doesn't seem to work on this chip unfortunately.
Would need to buy a different one. I just have a cheap one.

I have the 7840u I believe the 2023 edition with 64gb of ram.

**Reply by astearon** [reply]
Ok so at least i know the 2023 model is fixable when this kind of bios issue happens. 
What would really be great is for GPD to share with the community what to check on the board other than the bios chip when we have a broken bios or broken update/upload to the chip.

**Comment by mfoster978** [comment]
That's what I'm hoping for. I haven't heard back from her yet.

**Reply by astearon** [reply]
How much did you end up spending on shipping+repair? And where are you from if I may ask...some countries have cheaper transport to/from China

**Comment by mfoster978** [comment]
I don't actually have a link or program to that. It was another guy here
that mentioned it.

The reply I got back from Sophie was very limited she told me they only
replaced the bios chip nothing else but im really not 100 percent sure.

Also the shipping from the US cheapest option was about $60 the repair was
free. I want charged for shipping back to me.

**Reply by Torrengen** [reply]
interesting, if that was all that's needed (which it was for me), please share what the new chip's BIOS looks like when u get it back!

A little sad that they didn't give much more info, but still, huge props for repairing it for free though, even if the shipping wasn't free!
(Though knowing the distances involved, i'd be surprised if they'd ship it for free too, someone has to foot the bill afterall xD)

**Reply by astearon** [reply]
Just received 2 chips from GPD with the bios already on them. I did not test any as of yet but here is a dump!
https://mega.nz/file/PpsnybLJ#0JyE71fKqbZEIAhpcI-zeAsIM-yqoK8QERAmYcVtzhY

**Reply by astearon** [reply]
Chip replaced and I left it to charge a while.
Intake voltage shows 20v so at least it showing up....now if it actually takes a charge we'll see.
![IMG_20240629_220249](https://github.com/nofeletru/UsbAsp-flash/assets/7342330/09b2efe0-6d2e-4d82-86cc-131b0646571f)

**Reply by astearon** [reply]
In the end I had to send the unit back to them for repairs. The Amps never shot past 0.0X something.
Got the device back a couple of weeks ago and all is working now.

Most I can figure out is that the BIOS upgrade process messed more than just the BIOS Chip, perhaps something not mentioned but it would have been great if GPD gave us an actual fix process to follow.

**Comment by mfoster978** [comment]
Yeah I agree with you. Sadly I had to do the same thing. GPD did fix mine
for free although I did have to pay shipping to get it to them and it took
a few months to go back. Either way it's been working great for months now
and I've learned to never update the bios 😂

**Reply by astearon** [reply]
They repaired mine for free as well, payed for transport a bunch though. Mine shipped to China in 4 days, stayed in their queue for a week and back to me in another 4 days.
My conclusion is the same.... keep away from the BIOS upgrades!

**Comment by mfoster978** [comment]
Damn you got lucky! It took 3 months for me and the shipping cost me about
$120

**Comment by mfoster978** [comment]
I had the same problem which is why I ended up shipping my GPD in for
repairs. Took them a couple months but I got it back.

**Reply by trillox9** [reply]
I have a mini PC, support isn't answering me, I've corrupted the BIOS and I don't know what to do. Now I'm waiting for a WSON8 8*6 and I have to try again with NeoProgrammer, but I have little hope...

**Reply by astearon** [reply]
> I have a mini PC, support isn't answering me, I've corrupted the BIOS and I don't know what to do. Now I'm waiting for a WSON8 8*6 and I have to try again with NeoProgrammer, but I have little hope...

The chip is best read when taken off the motherboard, whilst still on the board there can be other components that will interact with the chip and corrupt the reading process.

**Reply by trillox9** [reply]
It's impossible for me to desolder it; moreover, it appears to be soldered differently on the motherboard, with the pins barely visible. Additionally, I would still need the proper software, and I don't know if there's support for the 25r256qjweq BIOS for the CH341B programmer.

**Reply by astearon** [reply]
If it is anything like the ones in the pictures from this thread (and by the name you've provided it looks like it is) then the chip has the pins underneath the chip, smt style. This means you need a hot air gun in order to take it off the board.

**Reply by Torrengen** [reply]
i was personally able to probe it on-board using a pogo-pin prober, it's finnicky, but saved me the risk of melting components ^^
it's possible to de-solder, but it has a massive heatsink pad on the bottom so u will need a hot-air gun, and even then it's not an easy process xD

**Comment by mfoster978** [comment]
I haven't really looked into any recent info on this but I'd imagine there
has to be a way to get software for these chips by now. As long as you have
the correct software for the chip you can definitely reprogram it.

**Comment by Dr1fte** [comment]
Did any of you guys manage to get a full read from the 25Q256JWEQ? 

I have the CH341A (Black) and was using the 1.8v adapter. But I could not get neoprogrammer to read. For reference, I removed the bios chip from the board. I inserted the chip into the clamp, ensuring to line up the pads against the prongs.

I’ve also tried soldering the chip on to the blank adapter board. And build the sandwich as it’s called. Still nothing through neoprogrammer. 

I’m using the neoprogrammer version mentioned by someone else in this git. Same for the drivers.

**Reply by ladmorfe** [reply]
tengo el mismo problema

**Comment by Dr1fte** [comment]
To the ones who have edited the BD790i bios. Could you answer me this,

Is it possible to put asus memory controller bios section into these boards to get control over trefi?

**Comment by Dr1fte** [comment]
For anyone interested. A WSON8 pogo 6x8 pinout will read/write and erase for you. You’ll have to use Neoprogrammer if you’re using a CH341A or B.

**Comment by kerneltoast** [comment]
FYI, I added support for W25R256JW to flashrom: https://review.coreboot.org/c/flashrom/+/90298
